fusion-cli/build/babel-plugins/babel-plugin-i18n/index.js

```diff
@@ -45,6 +49,42 @@ function i18nPlugin(babel /*: Object */, {translationIds} /*: PluginOpts */) {
             }
             translationIds.add(element.value);
           });
+        } else if (specifierName === 'useTranslations') {
+          if (!t.isVariableDeclarator(refPath.parentPath.parent)) {
+            throw new Error(
+              'Unexpected assignment of useTranslations return function'
+            );
+          }
+          const localName = refPath.parentPath.parent.id.name;
+          const translationPaths =
+            refPath.parentPath.scope.bindings[localName].referencePaths;
+          translationPaths.forEach(translationPath => {
+            if (
+              t.isCallExpression(translationPath.parentPath) &&
+              translationPath.parentKey === 'callee'
+            ) {
+              const arg = translationPath.parentPath.node.arguments[0];
+              const errorMessage =
+                'useTranslations result function must be passed string literal or hinted template literal';
+              if (t.isStringLiteral(arg)) {
+                translationIds.add(arg.value);
+              } else if (t.isTemplateLiteral(arg)) {
+                const result = arg.quasis.map(q => q.value.raw);
```

Comment:
Oh, it seems this coalescing already happens in some cases, but not all?

```
`abc${foo}${bar}def${baz}`;
// 4 elements ["abc", "", "def", ""]

`${foo}${bar}`
// 3 elements
["", "", ""]
```

Reply:
Yeah cooked is more accurate. Some examples:
```js
'a\bc' // => 'ac'
'a\\bc' // => 'a\bc'
'a\`bc' // => 'a`bc'
```

Reply:
I think the reason I didn't want to completely ignore empty strings was because it is significant if they either start or end the match (meaning no `^` or `$` in regex terms). I think empty strings could be coalesced into a single empty string (or removed if in the middle of a match).